### ISSUE TITLE
Improving the accessibility of qtype_matrix

### ISSUE THREAD

**Issue by edicthagen** [ndunand/moodle-qtype_matrix#76]
The overall accessibility of the matrix question type should be improved. It currently fails WCAG2.2 success criteria [1.3.1](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html), [1.3.2](https://www.w3.org/WAI/WCAG22/Understanding/meaningful-sequence) as well as [3.3.2](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions). Basic accessibility requirements for MC questions can also be found in a [blog post by Terrill Thompson](https://terrillthompson.com/854).
In the most straightforward setting of this question type, a sighted user sees four (or more) statements and has to decide whether each of these statements is either true or false. The statements and the radio buttons are placed within a table. In [this simplified example (JSFiddle)](https://jsfiddle.net/dua7v56z/latest/), the question consists of a table with three columns and five rows. The standard matrix question has several accessibility errors:

- Compared to a sighted user, for a screenreader user it is hard to determine which radio button is associated with which question and which answer. The radio buttons lack a clear label, either associated via <label> element or aria-labelledby.
- The  table layout is inconsistent, as the column headers are correctly marked as th, but the row headers are not. This makes it more complicated to navigate the table if using a screenreader.
- Though the user must answer each of the statements and mark them as either true or false, this is not indicated to the user via the required tag.
- In case the page is linearized, e.g. when a sighted user operates the browser differently when using a magnifying software, the information on which radio button is associated with which table column is lost.

![Image](https://github.com/user-attachments/assets/f98d73a0-0b7d-4d33-8edc-c1bd14a477c3)

I therefore created three example matrix questions with improved accessibility:

The [first (see this JSFiddle for full example)](https://jsfiddle.net/springtime_evaluator/ynkapsq1/1/) represents the mininum accessibility requirements with the following improvements
```
        <thead>
            <tr>
                <td></td>
                <th id="true" scope="col">True</th>
                <th id="false" scope="col">False</th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <th id="answer1" scope="row">This statement is false</th>
                <td><input type="radio" name="q562842:1_cell74558" value="32435" aria-labelledby="answer1 true"></td>
                <td><input type="radio" name="q562842:1_cell74558" value="32436" aria-labelledby="answer1 false"></td>
            </tr>
```

- th for each row with a statement
- th of cols/rows are used as labels for the corresponding radio button via "aria-labelledby". 
- required tag added to each radio button
- additional (short) instruction on how to complete this questions; could also be generated programmatically

The [second variant (see JSFiddle)](https://jsfiddle.net/springtime_evaluator/6h7dbx8r/1/) adds labels to each radio button and uses a aria-labelledby to associate the label as well as the row header with the radio buttons:

```
            <tr>
                <th id="answer1" scope="row">This statement is false</th>
                <td>
                    <label id="q562842:1_cell74558_a1" for="q562842:1_cell74558_1">True</label>
                    <input type="radio" name="q562842:1_cell74558" id="q562842:1_cell74558_1" value="32435" aria-labelledby="answer1 q562842:1_cell74558_a1" required>
                </td>
                <td>
                    <label id="q562842:1_cell74558_a2" for="q562842:1_cell74558_2">False</label>
                    <input type="radio" name="q562842:1_cell74558" id="q562842:1_cell74558_2" value="32436" aria-labelledby="answer1 q562842:1_cell74558_a2" required>
                </td>
            </tr>
```

The labels for each radio button make it easier to understand which answer the radio button is corresponds to. E.g. in case the page is linearized.

The [third and final variant (see JSFiddle)](https://jsfiddle.net/springtime_evaluator/s8zc970o/1/) uses a different approach. Instead of using a layout table this variant uses a fieldset/legend for each statement plus individual labels for each radio button:
```
    <fieldset>
        <legend>This statement is false</legend>
        <label for="q562842:1_cell74558_1">True</label><input type="radio" name="q562842:1_cell74558" id="q562842:1_cell74558_1" value="32435" required>
        <label for="q562842:1_cell74558_2">False</label><input type="radio" name="q562842:1_cell74558" id="q562842:1_cell74558_2" value="32436" required>
    </fieldset>
```
